packages/shortest/src/types/config.ts

```diff
@@ -1,32 +1,119 @@
 import { z } from "zod";
+import {
+  LLMSupportedModels,
+  LLMSupportedModelsType,
+  LLMSupportedProviders,
+  LLMSupportedProvidersType,
+} from "./ai";
+import { getDefaultProviderModel } from "@/ai/client-provider";
+import { MakeOptional } from "@/utils/types";
 
-export interface ShortestConfig {
-  headless: boolean;
-  baseUrl: string;
-  testPattern: string;
-  anthropicKey?: string;
-  mailosaur?: {
-    apiKey?: string;
-    serverId?: string;
-  };
-}
-
+/**
+ * Schema for Mailosaur configuration.
+ */
 const mailosaurSchema = z
   .object({
     apiKey: z.string(),
     serverId: z.string(),
   })
   .optional();
 
-export const configSchema = z
+export interface MailosaurConfig {
+  apiKey: string;
+  serverId: string;
+}
+
+/**
+ * Schema for the LLM (AI) configuration.
+ * If no model is provided, falls back to a default based on the provider.
+ */
+const llmSchema = z.object({
+  provider: z.literal(LLMSupportedProviders.ANTHROPIC),
+  apiKey: z.string(),
+  model: z
+    .enum([LLMSupportedModels.CLAUDE_3_5_SONNET])
+    .default(getDefaultProviderModel(LLMSupportedProviders.ANTHROPIC)),
+});
+export interface LLMConfig {
+  provider: LLMSupportedProviders;
+  apiKey: string;
+  model: LLMSupportedModels;
+}
+
+export interface LLMPPublicConfig {
+  provider: LLMSupportedProvidersType;
```

Comment:
I think even this one can be optional. Ideally, there should be zero config to run Shortest.

No `ai` prop provided should default to the provider we think is the best, e.g. `anthropic`, which tries to read the `apiKey` from ENV (as Vercel AI SDK supports it by default).